### PR TITLE
gateway: shut down tracer and logger providers in graceful shutdown

### PR DESCRIPTION
**Pull request by tomhoule**
So the exporters have time to flush during the grace period.

### CODE REVIEW

**Review by yoav-lavi — APPROVED** [review]
Xzibit: "so I put a graceful shutdown in your graceful shutdown so you can shutdown while you shutdown"

File: engine/crates/telemetry/src/graceful_shutdown.rs [hunk 1]
```diff
@@ -0,0 +1,4 @@
+pub fn graceful_shutdown() {
+    opentelemetry::global::shutdown_tracer_provider();
```
Comment (Finistere) [inline engine/crates/telemetry/src/graceful_shutdown.rs:2]: would need to shut down the meter provider but that's also only available through the "SDK" implementation, not the interface -_-"

Comment (Finistere) [reply engine/crates/telemetry/src/graceful_shutdown.rs:2]: can't even shutdown simultaneously lovely.

Comment (tomhoule) [reply engine/crates/telemetry/src/graceful_shutdown.rs:2]: Yes it's challenging to wire that into this function, but I could spend some more time and try harder.

Comment (Finistere) [reply engine/crates/telemetry/src/graceful_shutdown.rs:2]: Honestly you can probably leave it as is, I'll remove the reload logic, so can probably keep the SdkMeter ref somewhere

**Review by Finistere — APPROVED** [review]

**Pull request merged**